### ISSUE TITLE
[Filesystem Bug] Copy & pasting a folder into itself once actually pastes recursively

### ISSUE THREAD

**Issue by rwalle61** [theia-ide/theia#3454]
If you copy a folder and paste it into itself once, theia keeps pasting recursively.

E.g.:
![screen shot 2018-10-30 at 1 14 19 pm](https://media.github.ibm.com/user/186211/files/0d131b50-dcfd-11e8-9f81-f46b804f89b4)

The folder is copied recursively in the file system too, not just a UI glitch:
```bash-4.4# pwd
/microclimate-workspace/nodetest/server/config/config/config/config/config
```

Perhaps theia is reading and writing to the directory at the same time.

**Comment by simark** [comment]
Indeed.  The recursion seems to stop when we hit PATH_MAX (4096).  The absolute name of the most nested directory is 4095 chars long, in my test.

**Comment by simark** [comment]
`cp` is smart.  We should be more like `cp`.

```
cp -a ./doc ../theia/doc
cp: cannot copy a directory, './doc', into itself, '../theia/doc/doc'
```

**Comment by kumudhakn** [comment]
Hi All.. 
I am new to Theia and this is my first issue. I tried to fix this issue and raised a PR #3599.
Request you to review the same.

Thanks.

**Comment by rwalle61** [comment]
thanks @kumudhakn :)
